### PR TITLE
fix bad typings file caused by https://github.com/egoist/tsup/issues/782

### LINKED ISSUE #21820: [Bug]: 7.0.0-rc.8 Cannot find module 'lib/types/dist' in @storybook/html
### Describe the bug

In Storybook 7.0.0-rc.8 I see a TypeScript compile error when I try to build a project that imports types from `@storybook/html`:
```
../../node_modules/@storybook/html/dist/types-2ec88714.d.ts:3:33 - error TS2307: Cannot find module 'lib/types/dist' or its corresponding type declarations.

import * as lib_types_dist from 'lib/types/dist';
```
(There is a similar issue with `import * as lib_docs_tools_dist from 'lib/docs-tools/dist';`)

If I look in `node_modules` I don't see any directory for `lib` or any entry points named `lib` in `package.json`. Are these standard ES module imports? Is a certain bundler configuration required for this import to be found? If I locally modify `node_modules/@storybook/html/dist/types-2ec88714.d.ts` as described below, the build succeeds.

`import * as lib_types_dist from 'lib/docs-tools/dist';` change to 
`import * as lib_types_dist from '@storybook/docs-tools';`

### To Reproduce

Currently [failing the build](https://github.com/ni/nimble/actions/runs/4536747276/jobs/7993806135#step:9:92) in [this branch of the ni/nimble repository](https://github.com/ni/nimble/tree/users/jattas/storybook-7-plus-typescript). To build, clone that branch then run `npm i` and `npm run build`. The build for `/packages/nimble-components` will fail with this error.

If a smaller reproduction case would be helpful I can try to create one. 

### System

```shell
Environment Info:

  System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 16.15.1 - /usr/local/bin/node
    npm: 8.11.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 111.0.5563.110
    Firefox: 111.0.1
    Safari: 16.3
```

### Additional context

A workaround is to change `tsconfig.json` to include `"skipLibCheck": true`, which tells the TypeScript compiler not to run type checking on code in `node_modules`. But that impacts compile for all dependencies so we'd prefer not to have to do this.

### LINKED ISSUE #22288: [Bug]: upgrading to 7.0.7 breaks app start/compile
### Describe the bug

Running the `npx storybook@latest upgrade` upgraded to 7.0.7 but `@storybook/addon-actions` causes the following error (reverting just @storybook/addon-actions to 7.0.6 fixes the issue) when attempting to start/compile my app:

sb-addon

### To Reproduce

_No response_

### System

```shell
m087610:sts-react (develop) ✔ npx storybook@latest info

Environment Info:

  System:
    OS: macOS 13.3.1
    CPU: (8) arm64 Apple M1 Pro
  Binaries:
    Node: 20.0.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.4 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 112.0.5615.137
    Edge: 112.0.1722.58
    Safari: 16.4
  npmPackages:
    @storybook/addon-a11y: ^7.0.7 => 7.0.7 
    @storybook/addon-actions: 7.0.6 => 7.0.6 
    @storybook/addon-console: ^1.2.3 => 1.2.3 
    @storybook/addon-docs: ^7.0.7 => 7.0.7 
    @storybook/addon-essentials: ^7.0.7 => 7.0.7 
    @storybook/addon-interactions: ^7.0.7 => 7.0.7 
    @storybook/addon-links: ^7.0.7 => 7.0.7 
    @storybook/addon-mdx-gfm: ^7.0.7 => 7.0.7 
    @storybook/addon-viewport: ^7.0.7 => 7.0.7 
    @storybook/addons: ^7.0.7 => 7.0.7 
    @storybook/jest: ^0.1.0 => 0.1.0 
    @storybook/preset-create-react-app: ^7.0.7 => 7.0.7 
    @storybook/react: ^7.0.7 => 7.0.7 
    @storybook/react-webpack5: ^7.0.7 => 7.0.7 
    @storybook/test-runner: ^0.10.0 => 0.10.0 
    @storybook/testing-library: 0.1.0 => 0.1.0 
    @storybook/theming: ^7.0.7 => 7.0.7 
    @storybook/types: ^7.0.7 => 7.0.7
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ndelangen**
Closes https://github.com/storybookjs/storybook/issues/21820
Closes #22288

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/renderers/html/src/types.ts [hunk 1]
```diff
@@ -24,6 +24,19 @@ export interface HtmlRenderer extends WebRenderer {
   storyResult: StoryFnHtmlReturnType;
 }
 
+export interface Parameters extends DefaultParameters {
```
[the remark this answers is not in the document]
Comment (ndelangen) [reply code/renderers/html/src/types.ts:27]: we really don't. I'll revert this.

**Comment by ndelangen** [comment]
@shilman I think this is a critical bugfix

File: code/addons/backgrounds/src/preview.tsx [hunk 2]
```diff
@@ -1,8 +1,9 @@
+import type { Addon_DecoratorFunction } from '@storybook/types';
 import { withBackground } from './decorators/withBackground';
 import { withGrid } from './decorators/withGrid';
 import { PARAM_KEY } from './constants';
 
-export const decorators = [withGrid, withBackground];
+export const decorators: Addon_DecoratorFunction[] = [withGrid, withBackground];
```
Comment (kasperpeulen) [inline code/addons/backgrounds/src/preview.tsx:6]: I would just use `import type { DecoratorFunction } from '@storybook/types';`

**Review by kasperpeulen — APPROVED** [review]
LGTM

File: code/addons/backgrounds/src/preview.tsx (hunk 2 above)
Comment (ndelangen) [reply code/addons/backgrounds/src/preview.tsx:6]: These types are pretty much equal. When we do the types cleanup at 8.0 we'll sync this up.

**Pull request merged**